lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/DBODoiAssociationDaoImpl.java

```diff
@@ -98,6 +98,11 @@ public DoiAssociation createDoiAssociation(DoiAssociation dto) throws DuplicateK
 	@WriteTransaction
 	@Override
 	public DoiAssociation updateDoiAssociation(DoiAssociation dto) {
+		// Fill in the fields from the existing object that required for the DBO that the client may not pass in
+		DoiAssociation existing = getDoiAssociation(dto.getObjectId(), dto.getObjectType(), dto.getObjectVersion());
```

Comment:
Here we're making the second call to retrieve this DOI. Prior to this, the manager attempted to retrieve an eTag for the object.

I initially see two logical ways to avoid making this second call, and both involve avoiding the use of getEtagForUpdate:
- Remove this getDoiAssociation call and retrieve the entire association in the manager, instead of just the eTag. In this case, the DAO must assume the manager is setting all of the fields that should be immutable (like the ID of the association)
- Keep this getDoiAssociation call and compare the eTags in the DAO. The getDoiAssociation will throw a NotFoundException if it doesn't exist (thrown to the manager via this method) and this method can throw a ConflictingUpdateException if the eTags do not match. The manager then simply handles the exceptions as it does now, with the only change being the delegation the requirements of getEtagForUpdate to this method.